contracts/SynthetixRouterStrategy.sol

```diff
@@ -0,0 +1,351 @@
+        string memory _strategyName,
+        bytes32 _synth,
+        uint256 _susdBuffer
+    ) external returns (address newStrategy) {
+        require(isOriginal);
+        // Copied from https://github.com/optionality/clone-factory/blob/master/contracts/CloneFactory.sol
+        bytes20 addressBytes = bytes20(address(this));
+        assembly {
+            // EIP-1167 bytecode
+            let clone_code := mload(0x40)
+            mstore(
+                clone_code,
+                0x3d602d80600a3d3981f3363d3d373d3d3d363d73000000000000000000000000
+            )
+            mstore(add(clone_code, 0x14), addressBytes)
+            mstore(
+                add(clone_code, 0x28),
+                0x5af43d82803e903d91602b57fd5bf30000000000000000000000000000000000
+            )
+            newStrategy := create(0, clone_code, 0x37)
+        }
+
+        SynthetixRouterStrategy(newStrategy).initialize(
+            _vault,
+            _strategist,
+            _rewards,
+            _keeper,
+            _yVault,
+            _strategyName,
+            _synth,
+            _susdBuffer
+        );
+
+        emit FullCloned(newStrategy);
+    }
+
+    function initialize(
+        address _vault,
+        address _strategist,
+        address _rewards,
+        address _keeper,
+        address _yVault,
+        string memory _strategyName,
+        bytes32 _synth,
+        uint256 _susdBuffer
+    ) public {
+        super.initialize(
+            _vault,
+            _strategist,
+            _rewards,
+            _keeper,
+            _yVault,
+            _strategyName
+        );
+        _initializeSynthetixRouter(_synth, _susdBuffer);
+    }
+
+    function _initializeSynthetixRouter(bytes32 _synth, uint256 _susdBuffer)
+        internal
+    {
+        _initializeSynthetix(_synth);
+        susdBuffer = _susdBuffer;
+    }
+
+    function adjustPosition(uint256 _debtOutstanding) internal override {
+        if (emergencyExit) {
+            return;
+        }
+        uint256 looseSynth = _balanceOfSynth();
+        uint256 _sUSDBalance = _balanceOfSUSD();
+
+        // this will tell us how much we need to keep in the buffer
+        uint256 totalDebt = vault.strategies(address(this)).totalDebt; // in sUSD (want)
+        uint256 buffer = totalDebt.mul(susdBuffer).div(DENOMINATOR);
+        uint256 debtRatio = vault.strategies(address(this)).debtRatio;
+
+        uint256 _sUSDToInvest =
+            _sUSDBalance > buffer ? _sUSDBalance.sub(buffer) : 0;
+        uint256 _sUSDNeeded = _sUSDToInvest == 0 ? buffer.sub(_sUSDBalance) : 0;
+        uint256 _synthToSell =
+            _sUSDNeeded > 0 ? _synthFromSUSD(_sUSDNeeded) : 0; // amount of Synth that we need to sell to refill buffer
+        uint256 _synthToInvest =
+            looseSynth > _synthToSell ? looseSynth.sub(_synthToSell) : 0;
+
+        if (_synthToSell == 0 && debtRatio > 0) {
+            // This will invest all available sUSD (exchanging to Synth first)
```

Comment:
the logic is the following: if there is any loose synth and it is not locked (I added this check now), first deposit in vault and then do the exchange. If we do the exchange and there was remaining not deposited synth, it will also get locked. So first deposit any loose synth and then exchange. After this, we have to manually call depositInVault after settlement period.

Reply:
yep that is what I was trying to say. the comment says otherwise tho

thanks

Reply:
I changed the comment, too